src/pages/transaction/index.tsx

```diff
@@ -74,11 +75,19 @@ export const TransactionPage: React.FC = () => {
     return 'Sign transaction';
   }, [txType]);
 
+  useEffect(() => {
+    window.addEventListener('beforeunload', async function () {
+      if (!isSubmitting) {
+        await doSubmitOrCancelPendingTransaction('cancel');
+      }
+    });
```

Comment:
```suggestion
    window.addEventListener('beforeunload', async () => {
      if (!isSubmitting) {
        await doSubmitOrCancelPendingTransaction('cancel');
      }
    });
```
Nit, but pref. to use an anon function here, as you don't need a new scope

Reply:
Yeah, I agree on this. I was struggle to get the `useEffect` to work on close. I first tried a return clean up function that didn't work ...then copied this in as a suggestion to try. I prefer the anon func too. I'll change.